src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -694,16 +700,28 @@ end
   return max(v_mag_ll, v_mag_rr) + max(cf_ll, cf_rr)
 end
 
-
 @inline function max_abs_speed_naive(u_ll, u_rr, normal_direction::AbstractVector, equations::IdealGlmMhdEquations2D)
+  rho_ll, rho_v1_ll, rho_v2_ll, rho_v3_ll, _ = u_ll
+  rho_rr, rho_v1_rr, rho_v2_rr, rho_v3_rr, _ = u_rr
 
-  norm_ = norm(normal_direction)
-  # Normalize the vector without using `normalize` since we need to multiply by the `norm_` later
-  normal_vector = normal_direction / norm_
-  u_ll_rotated = rotate_to_x(u_ll, normal_vector, equations)
-  u_rr_rotated = rotate_to_x(u_rr, normal_vector, equations)
+  norm_squared = (normal_direction[1] * normal_direction[1] +
+                  normal_direction[2] * normal_direction[2])
+
+  # Calculate velocities and fast magnetoacoustic wave speeds
+  # left
+  v1_ll = rho_v1_ll / rho_ll
+  v2_ll = rho_v2_ll / rho_ll
+  v3_ll = rho_v3_ll / rho_ll
+  v_mag_ll = sqrt((v1_ll^2 + v2_ll^2 + v3_ll^2) * norm_squared)
```

Comment:
Similar question, would it be faster to use `v1_ll * v1_ll + v2_ll * v2_ll + v3_ll * v3_ll` here?

Reply:
Yes, done in e483df3b